### PR TITLE
Add database entity for parameters

### PR DESCRIPTION
**Pull request by shellderp**
Start dual writing to a new table. Next PR will be a self backfill which should be fun
Will also need another table for run_parameters

### CODE REVIEW

File: client/src/main/proto/app/cash/backfila/service.proto [hunk 1]
```diff
@@ -10,6 +10,7 @@ message Parameter {
   optional string name = 1;
   // Description to show in the UI.
   optional string description = 2;
+  optional bool required = 3;
```
Comment (shellderp) [inline client/src/main/proto/app/cash/backfila/service.proto:13]: Not sure how to wire this in the client yet. It's just so we can indicate in the web UI when a param is required

Comment (mpawliszyn) [reply client/src/main/proto/app/cash/backfila/service.proto:13]: It can just be part of the Annotation?

Comment (shellderp) [reply client/src/main/proto/app/cash/backfila/service.proto:13]: well in kotlin we know if it has a default or not, I just don't know where that is determined today. (won't be hard)

File: service/src/main/kotlin/app/cash/backfila/service/persistence/DbRegisteredBackfill.kt [hunk 2]
```diff
@@ -54,9 +57,14 @@ class DbRegisteredBackfill() : DbUnsharded<DbRegisteredBackfill>, DbTimestampedE
   override lateinit var updated_at: Instant
 
   // TODO(mgersh): this might want to be its own table
```
[the remark this answers is not in the document]
Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/service/persistence/DbRegisteredBackfill.kt:59]: when I remove it yep

**Pull request merged**